tools/automator/README.md

```diff
@@ -0,0 +1,116 @@
+| `AUTOMATOR_SRC_BRANCH` | Set to the value of Prow's [`PULL_BASE_REF`][prow jobs].                                                                                                                                          | `master`                                   |
+| `AUTOMATOR_SHA`        | Set to the git commit sha (if inside a git directory).                                                                                                                                            | `c6418d1439d12eab2f1a4eae89f6eee46a34c31b` |
+| `AUTOMATOR_SHA_SHORT`  | Set to the git commit sha trimmed to 8 character (if inside a git directory).                                                                                                                     | `c6418d14`                                 |
+
+### Examples
+
+- [Prow presubmit job][presubmit example]
+- [Prow postsubmit job][postsubmit example]
+- [Prow periodic job][periodic example]
+
+## Testing / Debugging
+
+To test or debug `automator.sh` locally, first ensure the execution environment meets the above [prerequisites](#prerequisites). Additionally, supply
+the `--dry-run` option. This will prevent all GitHub state modifications (i.e. fork, commit, and pull request). Instead, it will print a `git diff` to
+standard out if any files were changed, then exit.
+
+```bash
+/path/to/automator.sh \
+--org=istio \
+--repo=test-infra \
+--branch=master \
+--title='my automator test' \
+--cmd="echo 'do something fun' > file.txt" \
+--dry-run
+```
+
+```diff
+--- /dev/null
++++ b/file.txt
+@@ -0,0 +1 @@
++do something fun
+
+```
+
+[presubmit example]: https://github.com/istio/test-infra/blob/de701f31e850f48fc035a5ec383c2595661dcc7b/prow/cluster/jobs/istio/istio.io/istio.istio.io.master.gen.yaml#L253-L296
+[postsubmit example]: https://github.com/istio/test-infra/blob/f70e0ea292c56acb62c2ff6c7f618af119d4c67c/prow/cluster/jobs/istio/common-files/istio.common-files.master.gen.yaml#L31-L77
+[periodic example]: https://github.com/istio/test-infra/blob/de701f31e850f48fc035a5ec383c2595661dcc7b/prow/cluster/jobs/istio/istio.io/istio.istio.io.master.gen.yaml#L3-L46
+[prow jobs]: https://github.com/istio/test-infra/blob/de701f31e850f48fc035a5ec383c2595661dcc7b/prow/cluster/jobs/istio/istio.io/istio.istio.io.master.gen.yaml#L3-L46
+[github access token]: https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token
+[bash 4.0+]: https://www.tldp.org/LDP/abs/html/bashver4.html
+[git diff]: https://git-scm.com/docs/git-diff
+[`curl`]: https://curl.haxx.se/
+[`getopt`]: https://linux.die.net/man/1/getopt
+[`git`]: https://git-scm.com/
+[`jq`]: https://github.com/stedolan/jq
+[`pr-creator`]: https://github.com/kubernetes/test-infra/tree/master/robots/pr-creator
+[`reealpath`]: https://linux.die.net/man/1/realpath
```

Comment:
For Mac, I used: `brew install bash` (installs v5) and `brew install gnu-getopt`

Reply:
Another Mac issue is you get an error (sample command)
```
../test-infra/tools/automator/automator.sh --org istio --repo istio --dry-run --modifier=update_proxy_dep --cmd='bin/update_proxy.sh $AUTOMATOR_SHA'
/var/folders/jw/qff4nk3s6f97x5wht80t3w0r0000gn/T/ci-XXXXXXXXXX.dpEQYrKP ~/work/go/src/istio.io/proxy
date: illegal option -- I
usage: date [-jnRu] [-d dst] [-r seconds] [-t west] [-v[+|-]val[ymwdHMS]] ...
            [-f fmt date | [[[mm]dd]HH]MM[[cc]yy][.ss]] [+format]
```
again due to a Mac inconsistency. To fix, one needs to `brew install coreutils` and then alias _date_ to _gdate_.

Reply:
I will follow-up and add the Mac-specific tools to the README as well.